### PR TITLE
feat(playground): add leaf connection icons inside playground VSCODE-650

### PR DESCRIPTION
**Pull request by gagik**
This adds a script to generate an icon font from the SVGs we have and use them inside VSCode, specifically the playground connection codelens.

### CODE REVIEW

File: scripts/generate-icon-font.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/** Icons to include in the generated icon font */
+const INCLUDED_ICONS = ['connection-active', 'connection-inactive'];
+
+/**
+ * Generates an icon font from the included icons and outputs package.json
+ * configuration field for those icons as specified in
+ * https://code.visualstudio.com/api/references/icons-in-labels
+ */
+async function main(): Promise<void> {
+  const font = await webfont({
```
Comment (gagik) [inline scripts/generate-icon-font.ts:14]: I think it's worth it to have a external dev dependency / script for this. We can potentially utilize Vscode's support for icons fonts for i.e. easily having different color versions of the same icons and, as in this case, easily including icons in different places.
The link from the comment:
https://code.visualstudio.com/api/references/icons-in-labels

File: package.json [hunk 2]
```diff
@@ -1154,6 +1155,22 @@
           "description": "Specify a shell command that is run to start the browser for authenticating with the OIDC identity provider for the server connection. Leave this empty for default browser."
         }
       }
+    },
+    "icons": {
+      "connection-active": {
```
Comment (Anemy) [inline package.json:1160]: Should we prefix this with a MongoDB identifier? That way if there's ever a `connection-active` added to the long icon list https://code.visualstudio.com/api/references/icons-in-labels#icon-listing or by another extension we don't conflict.
Maybe `mongodb-connection-active`?

Comment (gagik) [reply package.json:1160]: agreed, it'd actually help with the JSON schema for this as VSCode always expects at least one `-`. Could `mdb-` be alright too to keep the strings with icons short?

Comment (Anemy) [reply package.json:1160]: `mdb-` sgtm!

**Comment by alenakhineika** [comment]
Maybe the style of icons would look more aligned if the `ai` icon also didn't have a color like the `connected` icon?

Screenshot 2024-11-27 at 14 26 46

**Comment by gagik** [comment]
> Maybe the style of icons would look more aligned if the `ai` icon also didn't have a color like the `connected` icon?
> Screenshot 2024-11-27 at 14 26 46

We discussed that with @GaurabAryal and figured color could be cool in terms of encouraging people to click on them for this specific case.

**Review by alenakhineika — APPROVED** [review]

**Review by Anemy — APPROVED** [review]

**Pull request merged**